selftests/unit/test_utils_network.py

```diff
@@ -33,5 +35,64 @@ def test_release_port(self):
         self.assertNotIn(22, tracker.retained_ports)
 
 
+def get_all_local_addrs():
+    """
+    Returns all ipv4/ipv6 addresses that are associated with this machine
+    """
+    ipv4_addrs = []
+    ipv6_addrs = []
+    for interface in netifaces.interfaces():
+        ifaddresses = netifaces.ifaddresses(interface)
+        ipv4_addrs += [_['addr']
+                       for _ in ifaddresses.get(netifaces.AF_INET, [])]
+        ipv6_addrs += [_['addr']
+                       for _ in ifaddresses.get(netifaces.AF_INET6, [])]
+    return ipv4_addrs, ipv6_addrs
+
+
+class FreePort(unittest.TestCase):
+
+    def test_is_port_free(self):
```

Comment:
This test is really long and complex. Wouldn't be possible to simplify it? I would recommend use some mock objects instead of getting real IP addresses and running socket for real. I think we don't need to test the Python code related to socket.

Reply:
I'd prefer this way because with the use of mocking we would only test our perception of ports handling, but for years it was wrong. And even my first version developed on system with IPv6 was way different than the one I tried with docker. Last when I ran it in travis it generated even more exceptions I needed to address, which only increases the need for a functional test, what do you think?

Btw what do you mean by long? It takes near to zero seconds on my machine...